### PR TITLE
Implement the userBatchesView service

### LINKED ISSUE #460: Get all user batches service
## Overview 
Group managers may be allowed to create a set of users in batch (for a contest, ...). 
And should be able to list the batches of users they have already created ([see the bottom on this designs](https://france-ioi.github.io/algorea-designs/01.Groups_02.Composition_d.Invitation-modes)).

On `GET /user_batches/by-group/{group_id}/`

## Input
* group_id (in url)

## Output
* A list of the batches whose the group of group_prefix is one the input group's ancestors ~~and which are "allow_new"~~:
  - group_prefix
  - custom_prefix
  - size
  - creator_id

## Validations
* The current user is manager of the ~~input.group~~ prefix group id (or its ancestors) with at least can_manage:memberships.

### PR DESCRIPTION
**Pull request by zenovich**
Fixes #460

### CODE REVIEW

**Comment by zenovich** [comment]
Note: the service doesn't filter on `user_batch_prefixes.allow_new` (I hope it is the correct behavior)

**Comment by smadbe** [comment]
> Note: the service doesn't filter on `user_batch_prefixes.allow_new` (I hope it is the correct behavior)

Yes sorry, I forgot to remove this at the end, sorry.

File: app/api/groups/get_user_batches.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	if !found {
+		return service.InsufficientAccessRightsError
+	}
+
+	query := srv.Store.UserBatches().
+		Joins("JOIN user_batch_prefixes USING(group_prefix)").
+		Joins(`
+			JOIN groups_ancestors_active
+				ON groups_ancestors_active.ancestor_group_id = ? AND
+				   groups_ancestors_active.child_group_id = user_batch_prefixes.group_id`, groupID).
```
Comment (smadbe) [inline app/api/groups/get_user_batches.go]: That should be the opposite 🤨

Comment (zenovich) [reply app/api/groups/get_user_batches.go]: fixed

File: app/api/groups/get_user_batches.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	// required: true
+	CreatorID *int64 `json:"creator_id"`
+}
+
+// swagger:operation GET /user-batches/by-group/{group_id} groups userBatchesView
+// ---
+// summary: List user batches
+// description: >
+//
+//   Lists existing batches of users.
```
[the remark this answers is not in the document]
Comment (zenovich) [reply app/api/groups/get_user_batches.go]: fixed

File: app/api/groups/get_user_batches.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+//
+//
+//   Restrictions:
+//
+//   * The authenticated user (or one of his group ancestors) should be a manager of the `group_id`
+//     (directly, or of one of its ancestors)
+//     with at least 'can_manage:memberships', otherwise the 'forbidden' response is returned.
```
[the remark this answers is not in the document]
Comment (zenovich) [reply app/api/groups/get_user_batches.go]: fixed

File: app/api/groups/get_user_batches.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+//                (`from.group_prefix` is required when `from.custom_prefix` is given)
+//   in: query
+//   type: string
+// - name: from.size
+//   description: Start the page from the batch next to the batch with `user_batches.size` = `from.size`
+//                (`from.group_prefix` & `from.custom_prefix` are required when `from.size` is given)
+//   in: query
+//   type: string
+// - name: limit
+//   description: Display the first N members
```
[the remark this answers is not in the document]
Comment (zenovich) [reply app/api/groups/get_user_batches.go]: fixed

**Pull request merged**